### PR TITLE
swss/orchagent : Fixed the crash on error returned by select.

### PR DESCRIPTION
**Pull request by garyachy**
Crash happens when the select call in OrchDaemon start method
returns an error and getOrchByConsumer returns a null pointer.

Signed-off-by: Denys Haryachyy <Denys.Haryachyy@caviumnetworks.com>

### CODE REVIEW

**Comment by eladraz** [comment]
Can you explain why crash happens? Why the select() fails with error?

**Comment by garyachy** [comment]
Segmentation fault happens as far as there is no check on null pointer.
Select returns with an error whenever a signal is received by the application.

**Comment by eladraz** [comment]
I see, when someone INT signal.
Isn't is better to do exit() or return instead of continue?

**Comment by stcheng** [comment]
continue works as long as we print out some error messages i think.

File: orchagent/orchdaemon.cpp [hunk 1]
```diff
@@ -51,7 +51,10 @@ void OrchDaemon::start()
 
         ret = m_select->select(&s, &fd, 1);
         if (ret == Select::ERROR)
+        {
             SWSS_LOG_NOTICE("Error!\n");
```
Comment (kcudnik) [inline orchagent/orchdaemon.cpp]: Add more information about what error? And please remove \n from the end of log

Comment (garyachy) [inline orchagent/orchdaemon.cpp]: Added errno output that will help identify the reason of an error.

**Comment by garyachy** [comment]
I have fixed error message. Please merge.

**Comment by stcheng** [comment]
ack

**Comment by stcheng** [comment]
thanks

**Pull request closed without merging**